src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberFieldSymbolFromDesignation.cs

```diff
@@ -19,24 +19,29 @@ internal class SourceMemberFieldSymbolFromDesignation : SourceMemberFieldSymbol
         internal SourceMemberFieldSymbolFromDesignation(
             SourceMemberContainerTypeSymbol containingType,
             SingleVariableDesignationSyntax designation,
+            TypeSyntax typeSyntax,
             DeclarationModifiers modifiers)
             : base(containingType, modifiers, designation.Identifier.ValueText, designation.GetReference(), designation.Identifier.GetLocation())
         {
             Debug.Assert(DeclaredAccessibility == Accessibility.Private);
+            TypeSyntax = typeSyntax;
```

Comment:
>TypeSyntax = typeSyntax; [](start = 12, length = 24)

Field symbols should not hold on to syntax nodes directly, please use syntax reference as internal data instead.